src/LibraryManager/Manifest.cs

```diff
@@ -127,6 +127,210 @@ private static bool IsValidManifestVersion(string version)
             }
         }
 
+        /// <summary>
+        /// Installs a library with the given libraryId and 
+        /// </summary>
+        /// <param name="libraryId"></param>
+        /// <param name="providerId"></param>
+        /// <param name="files"></param>
+        /// <param name="destination"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        public async Task<ILibraryInstallationResult> InstallLibraryAsync(string libraryId, string providerId, IReadOnlyList<string> files, string destination, CancellationToken cancellationToken)
+        {
+            ILibraryInstallationResult result = null;
+            var errors = new List<IError>();
+            if (string.IsNullOrEmpty(libraryId))
+            {
+                errors.Add(PredefinedErrors.LibraryIdIsUndefined());
+            }
+
+            if (string.IsNullOrEmpty(destination) && string.IsNullOrEmpty(DefaultDestination))
+            {
+                errors.Add(PredefinedErrors.PathIsUndefined());
+            }
+            else
+            {
+                destination = string.IsNullOrEmpty(destination) ? DefaultDestination : destination;
+            }
+
+            IProvider provider = null;
+            if (string.IsNullOrEmpty(providerId) && string.IsNullOrEmpty(DefaultProvider))
+            {
+                errors.Add(PredefinedErrors.ProviderIsUndefined());
+            }
+            else
+            {
+                providerId = string.IsNullOrEmpty(providerId) ? DefaultProvider : providerId;
+
+                provider = _dependencies.Providers.FirstOrDefault(p => p.Id == providerId);
+                if (provider == null)
+                {
+                    errors.Add(PredefinedErrors.ProviderUnknown(providerId));
+                }
+            }
+
+            if (errors.Any())
+            {
+                result = new LibraryInstallationResult(errors);
+                return result;
+            }
+
+            var desiredState = new LibraryInstallationState()
+            {
+                LibraryId = libraryId,
+                Files = files,
+                ProviderId = providerId,
+                DestinationPath = destination
+            };
+
+            if (CheckAlreadyInstalled(desiredState))
```

Comment:
This only checks if the library specification is already in the manifest but does not actually install. It would fail to refresh libraries that are newer in the cache than the one in the project. I have implemented a IsLibraryUpToDate check and it would return a LibraryInstallationResult withe new property UpToDate set to true.

Reply:
Take a look at https://github.com/aspnet/LibraryManager/blob/dev/raquela/StressAndPerfFixes/src/LibraryManager/Manifest.cs RestoreLibraryAsync is probably what you want to call here. I can make it public.

Reply:
The provider checks weather or not the library is up to date or not

Reply:
This check is to prevent user from installing same library more than once to the same location.

Reply:
do we want to error for that? This can break a build is user sets restore on build. Should we mark it as a warning, We can try to add support for that later.

Reply:
This is a new API I added explicitly for the libman install command. This isn't the same one called by during the Restore operation.